contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -66,16 +86,41 @@ contract TrueCreditAgency is UpgradeableClaimable {
         emit PoolAllowed(pool, isAllowed);
     }
 
-    function creditScoreAdjustmentRate(address borrower) public view returns (uint256) {
-        uint8 score = creditOracle.getScore(borrower);
+    function updateCreditScore(ITrueFiPool2 pool, address borrower) external {
+        uint8 oldScore = creditScore[pool][borrower];
+        uint8 newScore = creditOracle.getScore(borrower);
+        if (oldScore == newScore) {
+            return;
+        }
+        require(newScore > 0, "TrueCreditAgency: Cannot set credit score to 0");
+
+        _rebucket(pool, borrower, oldScore, newScore, borrowed[pool][borrower]);
+    }
+
+    function creditScoreAdjustmentRate(ITrueFiPool2 pool, address borrower) public view returns (uint256) {
+        return _creditScoreAdjustmentRate(creditScore[pool][borrower]);
+    }
+
+    function _creditScoreAdjustmentRate(uint8 score) internal view returns (uint256) {
         if (score == 0) {
             return type(uint256).max;
         }
         return creditAdjustmentCoefficient.mul(MAX_CREDIT_SCORE - score).div(score);
     }
 
+    function interest(ITrueFiPool2 pool, address borrower) external view returns (uint256) {
+        CreditScoreBucket storage bucket = buckets[pool][creditScore[pool][borrower]];
+        return _interest(pool, bucket, borrower);
+    }
+
     function borrow(ITrueFiPool2 pool, uint256 amount) external onlyAllowedBorrowers {
         require(isPoolAllowed[pool], "TrueCreditAgency: The pool is not whitelisted for borrowing");
+        uint8 oldScore = creditScore[pool][msg.sender];
+        uint8 newScore = creditOracle.getScore(msg.sender);
+        require(newScore > 0, "TrueCreditAgency: Cannot borrow without a credit score");
```

Comment:
There should be no need to manually restrict this as long as the adjusted credit limit is 0 and interest rate is max uint256

Reply:
Giving max uint256 resulted in overflow when we added premium rate a bit further. Caped it with 500%, should not cause problems now